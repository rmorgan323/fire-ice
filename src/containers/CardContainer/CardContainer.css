.card-container-component {
  display: flex;
  flex-wrap: wrap;
  width: 960px;
  margin: auto;
}

@media screen and (max-width: 959px) {
  .card-container-component {
    width: 640px;
  }
}

@media screen and (max-width: 639px) {
  .card-container-component {
    width: 320px;
  }
}