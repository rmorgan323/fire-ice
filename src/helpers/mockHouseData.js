const mockHouseData = [
  {
    name: "House Hightower of the Hightower",
    founded: "Age of Dawn",
    seats: ["The Hightower, Oldtown"],
    titles: ["Lord of the Hightower", "Lord of the Port", "Voice of Oldtown", "Defender of the Citadel", "Beacon of the South", "King of the High Tower (formerly)"],
    coatOfArms: "Cendr√©e, a tower argent with a beacon on fire gules",
    ancestralWeapons: ["Vigilance"],
    words: "We Light the Way"
  }
]

export default mockHouseData;