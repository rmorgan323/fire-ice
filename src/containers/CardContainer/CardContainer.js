import React, { Component } from 'react';
import PropTypes, { shape, func, string } from 'prop-types';
import './CardContainer.css';
import { connect } from 'react-redux';
import Card from '../Card/Card';

class CardContainer extends Component {
  constructor() {
    super();
  }

  getCleanCardData = () => {
    const houses = this.props.houses.reduce((houseArray, house) => {
      houseArray.push({
        name: house.name,
        founded: house.founded,
        seats: house.seats.join(', '),
        titles: house.titles.join(', '),
        coatOfArms: house.coatOfArms,
        ancestralWeapons: house.ancestralWeapons.join(', '),
        words: house.words
      })

      return houseArray;
    }, [])
    return houses
  }

  displayCards = () => {
    const houseData = this.getCleanCardData();
    console.log(houseData)

    const display = houseData.map((data, index) => {
      return <Card 
        name={data.name}
        founded={data.founded}
        seats={data.seats}
        titles={data.titles}
        coatOfArms={data.coatOfArms}
        ancestralWeapons={data.ancestralWeapons}
        words={data.words}
        key={index}
      />;
    })

    return display;
  }

  render() {
    return (
      <div>
        {this.displayCards()}
      </div>
    )
  }
}

const mapStateToProps = store => ({
  houses: store.houses
})

export default connect(mapStateToProps, null)(CardContainer);