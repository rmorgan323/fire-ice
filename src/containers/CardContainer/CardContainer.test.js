import React from 'react';
import { CardContainer, mapStateToProps } from './CardContainer';
import { shallow } from 'enzyme';
import mockHouseData from '../../helpers/mockHouseData';

describe('card container tests', () => {
  it('should match the snapshot when no houses are in props', () => {
    const renderedApp = shallow(<CardContainer houses={[]} />);

    expect(renderedApp).toMatchSnapshot();
  })

  it('should match the snapshot when houses are in props', () => {
    const renderedApp = shallow(<CardContainer houses={mockHouseData} />);

    expect(renderedApp).toMatchSnapshot();
  })
})

describe('mapStateToProps tests', () => {
  it('should pull houses from store', () => {
    const mockStore = { mockHouseData };
    const result = mapStateToProps(mockStore);

    expect(result.houses).toEqual(mockStore.houses);
  })
})


