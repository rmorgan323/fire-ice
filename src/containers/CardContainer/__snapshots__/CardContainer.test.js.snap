// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card container tests should match the snapshot when houses are in props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    houses={
        Array [
            Object {
              "ancestralWeapons": Array [
                "Vigilance",
              ],
              "coatOfArms": "Cendrée, a tower argent with a beacon on fire gules",
              "founded": "Age of Dawn",
              "name": "House Hightower of the Hightower",
              "seats": Array [
                "The Hightower, Oldtown",
              ],
              "titles": Array [
                "Lord of the Hightower",
                "Lord of the Port",
                "Voice of Oldtown",
                "Defender of the Citadel",
                "Beacon of the South",
                "King of the High Tower (formerly)",
              ],
              "words": "We Light the Way",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [
          <Card
            ancestralWeapons="Vigilance"
            coatOfArms="Cendrée, a tower argent with a beacon on fire gules"
            founded="Age of Dawn"
            name="House Hightower of the Hightower"
            seats="The Hightower, Oldtown"
            titles="Lord of the Hightower, Lord of the Port, Voice of Oldtown, Defender of the Citadel, Beacon of the South, King of the High Tower (formerly)"
            words="We Light the Way"
/>,
        ],
        undefined,
      ],
      "className": "card-container-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "ancestralWeapons": "Vigilance",
          "coatOfArms": "Cendrée, a tower argent with a beacon on fire gules",
          "founded": "Age of Dawn",
          "name": "House Hightower of the Hightower",
          "seats": "The Hightower, Oldtown",
          "titles": "Lord of the Hightower, Lord of the Port, Voice of Oldtown, Defender of the Citadel, Beacon of the South, King of the High Tower (formerly)",
          "words": "We Light the Way",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [
            <Card
              ancestralWeapons="Vigilance"
              coatOfArms="Cendrée, a tower argent with a beacon on fire gules"
              founded="Age of Dawn"
              name="House Hightower of the Hightower"
              seats="The Hightower, Oldtown"
              titles="Lord of the Hightower, Lord of the Port, Voice of Oldtown, Defender of the Citadel, Beacon of the South, King of the High Tower (formerly)"
              words="We Light the Way"
/>,
          ],
          undefined,
        ],
        "className": "card-container-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "ancestralWeapons": "Vigilance",
            "coatOfArms": "Cendrée, a tower argent with a beacon on fire gules",
            "founded": "Age of Dawn",
            "name": "House Hightower of the Hightower",
            "seats": "The Hightower, Oldtown",
            "titles": "Lord of the Hightower, Lord of the Port, Voice of Oldtown, Defender of the Citadel, Beacon of the South, King of the High Tower (formerly)",
            "words": "We Light the Way",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`card container tests should match the snapshot when no houses are in props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    houses={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        Array [],
        <Loading />,
      ],
      "className": "card-container-component",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          Array [],
          <Loading />,
        ],
        "className": "card-container-component",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
